### PR TITLE
raftstore-v2: support online change lock write buffer limit

### PR DESCRIPTION
**Pull request by SpadeA-Tang**
### What is changed and how it works?

Issue Number: Ref #14320

What's Changed:

```commit-message
support online change lock write buffer limit
```

Also, add some tests for change write buffer size.

image

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/engine_rocks/src/cf_options.rs [hunk 1]
```diff
@@ -40,6 +40,23 @@ impl RocksCfOptions {
     pub fn into_raw(self) -> RawCfOptions {
         self.0
     }
+
+    pub fn set_flush_size(&mut self, f: usize) -> Result<()> {
```
Comment (bufferflies) [inline components/engine_rocks/src/cf_options.rs:44]:
```suggestion
    pub fn set_flush_size(&mut self, flush_size: usize) -> Result<()> {
```

File: components/engine_rocks/src/cf_options.rs [hunk 2]
```diff
@@ -40,6 +40,23 @@ impl RocksCfOptions {
...
+        if let Some(m) = self.0.get_write_buffer_manager() {
+            m.set_flush_size(f);
+        } else {
+            return Err(box_err!("write buffer manager not found"));
```
Comment (bufferflies) [inline components/engine_rocks/src/cf_options.rs:48]: Do you need to know the cf name?

File: src/config/configurable.rs [hunk 3]
```diff
@@ -171,6 +177,17 @@ impl ConfigurableDb for TabletRegistry<RocksEngine> {
         })
     }
 
+    fn set_cf_flush_size(&self, cf: &str, f: usize) -> ConfigRes {
+        loop_registry(self, |cache| {
```
Comment (bufferflies) [inline src/config/configurable.rs:181]: Do you need to limit the size range?

File: components/engine_rocks/src/cf_options.rs (hunk 1 above)
Comment (SpadeA-Tang) [reply components/engine_rocks/src/cf_options.rs:44]: `f` is used for consistency as other `set_flush_size` use `f`.

File: src/config/configurable.rs (hunk 3 above)
Comment (SpadeA-Tang) [reply src/config/configurable.rs:181]: It's not limited before for global write buffer manager.

File: components/engine_rocks/src/cf_options.rs (hunk 2 above)
Comment (SpadeA-Tang) [reply components/engine_rocks/src/cf_options.rs:48]: I think is unnecessary, as the client should know the cf. And seems we cannot fetch the cf name here.

**Review by overvenus — APPROVED** [review]

**Review by bufferflies — APPROVED** [review]

**Pull request merged**